Bazel is currently broken on Darwin

**Describe the bug**

Bazel is currently broken on the latest master, tested at b30286c4151c4a055cf4383cff920023eb1d95da.

Bazel was built fine back at cda852a528824f8169eb8a48e9f44dc3a3637ab3. I'm currently bisecting trying to find the offending commit. 

**To Reproduce**
Steps to reproduce the behavior:
1. nix-build . -A bazel

**Expected behavior**
Bazel builds correctly on Darwin.

**Additional context**

<details><summary>Error log</summary>

```
ERROR: /private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-bazel-3.3.1.drv-0/bazel_src/src/main/tools/BUILD:5:10: Linking of rule '//src/main/tools:daemonize' failed (Exit 1): cc_wrapper.sh failed: error executing command 
  (cd /private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-bazel-3.3.1.drv-0/bazel_0IWweLEG/out/execroot/io_bazel && \
  exec env - \
    PATH=/nix/store/j7f1x6n60lhbnviw98h6nxmy07hsc3s8-zip-3.0/bin:/nix/store/xs2039rla1yc3sigz9mird3mc3i04md5-python3-3.8.3/bin:/nix/store/9qzjvlnjgdnrs0yy7djxn53h0715d9iz-unzip-6.0/bin:/nix/store/f39xqq56lv4pn7p9vwqdqc3d8cs84xjs-which-2.21/bin:/nix/store/rrddr39j8mbqzinyvj0fbh6jl9r4gqr1-bash/bin:/nix/store/1dgdim74d05ypll85vslm8i7kgzq78vw-cctools-port/bin:/nix/store/9dy3zhjs98g0v6dx7agxcrffw7znl3rm-clang-wrapper-6.0.1/bin:/nix/store/8cwl110l46354n7f2lm3hsznjl2vn2pd-clang-6.0.1/bin:/nix/store/riiv88a31ialydg81sbyhdb2gxibk7rp-coreutils-8.31/bin:/nix/store/k16kbqwwkqh9qqkj78x6k6f90h34z3w5-cctools-binutils-darwin-wrapper-927.0.2/bin:/nix/store/gaqhnwqcrlvzcmjaynbrvcci99s4bnym-cctools-binutils-darwin-927.0.2/bin:/nix/store/riiv88a31ialydg81sbyhdb2gxibk7rp-coreutils-8.31/bin:/nix/store/qs9l9iv5pvr3r7mm54mbxhylnfz7ai10-zulu1.8.0_202-8.36.0.1/bin:/nix/store/xs2039rla1yc3sigz9mird3mc3i04md5-python3-3.8.3/bin:/nix/store/riiv88a31ialydg81sbyhdb2gxibk7rp-coreutils-8.31/bin:/nix/store/sbjhrk7kcvq0192yfcjj7l27n3i9a30r-findutils-4.7.0/bin:/nix/store/9nxhnchdrdx0rpyy4qi6xmaiwdk9lk4n-diffutils-3.7/bin:/nix/store/4di8j96dbnh217f3j8hbymn95cbjldrn-gnused-4.8/bin:/nix/store/y3xjm08adqsyyqr3avrv2phf3dg24z6v-gnugrep-3.4/bin:/nix/store/ryhl8basqip5m1xlw8djxf8955piy935-gawk-5.1.0/bin:/nix/store/xsn9w4j884q2il33azyqa1azwn6nn8fy-gnutar-1.32/bin:/nix/store/pkbnsq80igh639lkr1sc02gja91hrlrw-gzip-1.10/bin:/nix/store/l6sqnr8jybdrcxj5pncj3hmjv14277l2-bzip2-1.0.6.0.1-bin/bin:/nix/store/5d351276qm80bvjs5g342grfv95bbk7a-gnumake-4.3/bin:/nix/store/knd85yc7iwli8344ghav3zli8d9gril0-bash-4.4-p23/bin:/nix/store/5z1bky2k4wvs7pq8is4k2998sm5vhvy0-patch-2.7.6/bin:/nix/store/crjss4ardjpcl0d97hi5dd98f4wcwxdy-xz-5.2.5-bin/bin:/nix/store/knd85yc7iwli8344ghav3zli8d9gril0-bash-4.4-p23/bin:/nix/store/riiv88a31ialydg81sbyhdb2gxibk7rp-coreutils-8.31/bin:/nix/store/sbjhrk7kcvq0192yfcjj7l27n3i9a30r-findutils-4.7.0/bin:/nix/store/ryhl8basqip5m1xlw8djxf8955piy935-gawk-5.1.0/bin:/nix/store/y3xjm08adqsyyqr3avrv2phf3dg24z6v-gnugrep-3.4/bin:/nix/store/fl8agfpvxwv4s22gf48p59ipdvw66c69-gnutar-1.32/bin:/nix/store/4di8j96dbnh217f3j8hbymn95cbjldrn-gnused-4.8/bin:/nix/store/pkbnsq80igh639lkr1sc02gja91hrlrw-gzip-1.10/bin:/nix/store/f39xqq56lv4pn7p9vwqdqc3d8cs84xjs-which-2.21/bin:/nix/store/9qzjvlnjgdnrs0yy7djxn53h0715d9iz-unzip-6.0/bin:/nix/store/00vf73cy2zb9da4f14hvwksczqy7bj2w-file-5.39/bin:/nix/store/j7f1x6n60lhbnviw98h6nxmy07hsc3s8-zip-3.0/bin \
    PWD=/proc/self/cwd \
  external/local_config_cc/cc_wrapper.sh -o bazel-out/darwin-opt/bin/src/main/tools/daemonize bazel-out/darwin-opt/bin/src/main/tools/_objs/daemonize/daemonize.o -Wl,-L/nix/store/xs2039rla1yc3sigz9mird3mc3i04md5-python3-3.8.3/lib -Wl,-L/nix/store/ph994by7ljdy8pqxi65hf1igg31i3yia-libc++-7.1.0/lib -Wl,-L/nix/store/wh3sbv8fayjrxzsirqd7q542z2vkh8bq-objc4-osx-10.12.6/lib -Wl,-L/nix/store/wrjx3kg268ff2iz0s1pczvjmra53jckk-libc++-6.0.1/lib -Wl,-L/nix/store/8rfqw276sgpnbf04057syr4iy86swvgr-libc++abi-6.0.1/lib -Wl,-L/nix/store/v792kxaqpqzxh6rwshh3kawx5mlmjy0h-compiler-rt-6.0.1/lib -Wl,-L/nix/store/xs2039rla1yc3sigz9mird3mc3i04md5-python3-3.8.3/lib -Wl,-L/nix/store/ph994by7ljdy8pqxi65hf1igg31i3yia-libc++-7.1.0/lib -Wl,-L/nix/store/wh3sbv8fayjrxzsirqd7q542z2vkh8bq-objc4-osx-10.12.6/lib -Wl,-L/nix/store/wrjx3kg268ff2iz0s1pczvjmra53jckk-libc++-6.0.1/lib -Wl,-L/nix/store/8rfqw276sgpnbf04057syr4iy86swvgr-libc++abi-6.0.1/lib -Wl,-L/nix/store/v792kxaqpqzxh6rwshh3kawx5mlmjy0h-compiler-rt-6.0.1/lib -Wl,-F/nix/store/9isvlk3d0gwkp7djdjk8s30qqxj0dxhj-apple-framework-CoreFoundation/Library/Frameworks -Wl,-F/nix/store/6xnm7dd889h66dpx0ba24ga2xi5n9qg4-apple-framework-CoreServices/Library/Frameworks -Wl,-F/nix/store/fy65hvlgyy44wbrmqjbpwk1mhn3glpss-apple-framework-Foundation/Library/Frameworks -undefined dynamic_lookup -headerpad_max_install_names -lstdc++ -lm)
Execution platform: //:default_host_platform
[46 / 161] 7 actions running
    Compiling external/bazel_tools/.../util/logging.cc [for host]; 1s local
    Compiling external/bazel_tools/.../ijar/zip_main.cc [for host]; 1s local
    @bazel_tools//third_party/ijar:zip; 1s local
    @bazel_tools//third_party/ijar:platform_utils; 1s local
    Compiling external/bazel_tools/.../ijar/zip.cc [for host]; 1s local
    @bazel_tools//src/main/cpp/util:errors; 0s local
    Compiling external/bazel_tools/.../util/strings.cc [for host]; 0s local
ld: library not found for -lstdc++
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
[46 / 161] 7 actions running
    Compiling external/bazel_tools/.../util/logging.cc [for host]; 1s local
    Compiling external/bazel_tools/.../ijar/zip_main.cc [for host]; 1s local
    @bazel_tools//third_party/ijar:zip; 1s local
    @bazel_tools//third_party/ijar:platform_utils; 1s local
    Compiling external/bazel_tools/.../ijar/zip.cc [for host]; 1s local
    @bazel_tools//src/main/cpp/util:errors; 0s local
    Compiling external/bazel_tools/.../util/strings.cc [for host]; 0s local
Target //src:bazel_nojdk failed to build
[53 / 161] checking cached actions
INFO: Elapsed time: 13.594s, Critical Path: 1.89s
[53 / 161] checking cached actions
INFO: 7 processes: 7 local.
[53 / 161] checking cached actions
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully

ERROR: Could not build Bazel
builder for '/nix/store/w7vnarbl63i0irldq2rmvnvyz6vq908l-bazel-3.3.1.drv' failed with exit code 1
error: build of '/nix/store/w7vnarbl63i0irldq2rmvnvyz6vq908l-bazel-3.3.1.drv' failed
```

</details>

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@Profpatsch @LnL7 @avdv @mboes 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: bazel
# a list of nixos modules affected by the problem
module:
```

> I suppose this might be related: [f3f7612](https://github.com/NixOS/nixpkgs/commit/f3f7612a409e4ebbec634ab7f20e094f206d396a)

It's possible. I reverted 704daf7a6e4e5f26c5b0ad67ed8bd1b2090aa8ed, 1085403acdbaa7337d6074074f69ec6698226474, and f3f7612a409e4ebbec634ab7f20e094f206d396a but I could not verify that it works because gcc failed to build.

<details><summary>gcc error</summary>

```
make[3]: Entering directory '/private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/build/x86_64-apple-darwin/libstdc++-v3/include'
echo timestamp > stamp-pb                                                           
echo timestamp > stamp-host                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
make[3]: [Makefile:1773: x86_64-apple-darwin/bits/largefile-config.h] Error 1 (ignored)
make[3]: [Makefile:1774: x86_64-apple-darwin/bits/largefile-config.h] Error 1 (ignored)
echo 0 > stamp-namespace-version               
echo 1 > stamp-visibility                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
echo 1 > stamp-extern-template                        
echo 1 > stamp-dual-abi                               
echo 1 > stamp-cxx11-abi              
echo 1 > stamp-allocator-new                                                                                              
echo 'define _GLIBCXX_USE_FLOAT128 1' > stamp-float128
sed -e '/^#pragma/b' \                                                                                                    
    -e '/^#/s/\([ABCDEFGHIJKLMNOPQRSTUVWXYZ_][ABCDEFGHIJKLMNOPQRSTUVWXYZ_]*\)/_GLIBCXX_\1/g' \
    -e 's/_GLIBCXX_SUPPORTS_WEAK/__GXX_WEAK__/g' \    
    -e 's/_GLIBCXX___MINGW32_GLIBCXX___/__MINGW32__/g' \
    -e 's,^#include "\(.*\)",#include <bits/\1>,g' \                                                                      
    < /private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/gcc-9.3.0/libstdc++-v3/../libgcc/gthr.h > x86_64-apple-darwin/bits/gthr.h
sed -e 's/\(UNUSED\)/_GLIBCXX_\1/g' \                                                                                     
    -e 's/\(GCC[ABCDEFGHIJKLMNOPQRSTUVWXYZ_]*_H\)/_GLIBCXX_\1/g' \
    < /private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/gcc-9.3.0/libstdc++-v3/../libgcc/gthr-single.h > x86_64-apple-darwin/bits/gthr-single.h
sed -e 's/\(UNUSED\)/_GLIBCXX_\1/g' \   
    -e 's/\(GCC[ABCDEFGHIJKLMNOPQRSTUVWXYZ_]*_H\)/_GLIBCXX_\1/g' \                                                        
    -e 's/SUPPORTS_WEAK/__GXX_WEAK__/g' \          
    -e 's/\([ABCDEFGHIJKLMNOPQRSTUVWXYZ_]*USE_WEAK\)/_GLIBCXX_\1/g' \                                                     
    < /private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/gcc-9.3.0/libstdc++-v3/../libgcc/gthr-posix.h > x86_64-apple-darwin/bits/gthr-posix.h
sed -e 's/\(UNUSED\)/_GLIBCXX_\1/g' \            
    -e 's/\(GCC[ABCDEFGHIJKLMNOPQRSTUVWXYZ_]*_H\)/_GLIBCXX_\1/g' \
    -e 's/SUPPORTS_WEAK/__GXX_WEAK__/g' \                   
    -e 's/\([ABCDEFGHIJKLMNOPQRSTUVWXYZ_]*USE_WEAK\)/_GLIBCXX_\1/g' \                                                     
    -e 's,^#include "\(.*\)",#include <bits/\1>,g' \                 
    < /private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/gcc-9.3.0/libstdc++-v3/../libgcc/gthr-posix.h > x86_64-apple-darwin/bits/gthr-default.h
make[3]: Leaving directory '/private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/build/x86_64-apple-darwin/libstdc++-v3/include'
make[2]: Leaving directory '/private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/build'
make[1]: *** [Makefile:20904: stage1-bubble] Error 2
make[1]: Leaving directory '/private/var/folders/60/dfpn4wfs3v7d68xvg1_r8h0w0000gq/T/nix-build-gcc-9.3.0.drv-0/build'
make: *** [Makefile:21209: bootstrap] Error 2                
builder for '/nix/store/nm6qkkh88b16cj4z53bi32qhvljbyh3z-gcc-9.3.0.drv' failed with exit code 2                           
building '/nix/store/crlbsvj8j46z721k0q9ip45cwnqymyxa-configd-osx-10.8.5.drv'...
cannot build derivation '/nix/store/hcqnixpmp2av1lc40w7k2kaagvv5k98p-remote_java_tools_darwin.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/cxjs5j15zpy2lx52m6jkrw36701cjpmp-bazel-rc.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/viv5s79j32mgnipgwk58bjkq361arwq6-bazel_rc.patch.drv': 1 dependencies couldn't be built 
cannot build derivation '/nix/store/gpp1xxnn2apr7ql1q7kn1h74mrwgwf3q-bazel-3.3.1.drv': 1 dependencies couldn't be built   
error: build of '/nix/store/gpp1xxnn2apr7ql1q7kn1h74mrwgwf3q-bazel-3.3.1.drv' failed
```

</details>

--- a/pkgs/development/tools/build-managers/bazel/bazel_0_26/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/bazel_0_26/default.nix
@@ -236,6 +236,8 @@ stdenv.mkDerivation rec {
       fetch --experimental_distdir=${distDir}
       build --copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt="/g')"
       build --host_copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt="/g')"
+      build --linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt="/g')"
+      build --host_linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt="/g')"
       build --linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt="-Wl,/g')"
       build --host_linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt="-Wl,/g')"
       build --host_javabase='@local_jdk//:jdk'
@@ -245,6 +247,8 @@ stdenv.mkDerivation rec {
       # add the same environment vars to compile.sh
       sed -e "/\$command \\\\$/a --copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --host_linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_javabase='@local_jdk//:jdk' \\\\" \

--- a/pkgs/development/tools/build-managers/bazel/bazel_0_29/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/bazel_0_29/default.nix
@@ -417,6 +417,8 @@ stdenv.mkDerivation rec {
       fetch --distdir=${distDir}
       build --copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt="/g')"
       build --host_copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt="/g')"
+      build --linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt="/g')"
+      build --host_linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt="/g')"
       build --linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt="-Wl,/g')"
       build --host_linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt="-Wl,/g')"
       build --host_javabase='@local_jdk//:jdk'
@@ -426,6 +428,8 @@ stdenv.mkDerivation rec {
       # add the same environment vars to compile.sh
       sed -e "/\$command \\\\$/a --copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --host_linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_javabase='@local_jdk//:jdk' \\\\" \

--- a/pkgs/development/tools/build-managers/bazel/bazel_1/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/bazel_1/default.nix
@@ -417,6 +417,8 @@ stdenv.mkDerivation rec {
       fetch --distdir=${distDir}
       build --copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt="/g')"
       build --host_copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt="/g')"
+      build --linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt="/g')"
+      build --host_linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt="/g')"
       build --linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt="-Wl,/g')"
       build --host_linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt="-Wl,/g')"
       build --host_javabase='@local_jdk//:jdk'
@@ -426,6 +428,8 @@ stdenv.mkDerivation rec {
       # add the same environment vars to compile.sh
       sed -e "/\$command \\\\$/a --copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --host_linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_javabase='@local_jdk//:jdk' \\\\" \

--- a/pkgs/development/tools/build-managers/bazel/bazel_3/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/bazel_3/default.nix
@@ -429,6 +429,8 @@ stdenv.mkDerivation rec {
       fetch --distdir=${distDir}
       build --copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt="/g')"
       build --host_copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt="/g')"
+      build --linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt="/g')"
+      build --host_linkopt="$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt="/g')"
       build --linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt="-Wl,/g')"
       build --host_linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt="-Wl,/g')"
       build --host_javabase='@local_jdk//:jdk'
@@ -438,6 +440,8 @@ stdenv.mkDerivation rec {
       # add the same environment vars to compile.sh
       sed -e "/\$command \\\\$/a --copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --linkopt=\"/g')\" \\\\" \
+          -e "/\$command \\\\$/a --host_linkopt=\"$(echo $(< ${stdenv.cc}/nix-support/libcxx-ldflags) | sed -e 's/ /" --host_linkopt=\"/g')\" \\\\" \
           -e "/\$command \\\\$/a --linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt=\"-Wl,/g')\" \\\\" \
           -e "/\$command \\\\$/a --host_javabase='@local_jdk//:jdk' \\\\" \

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -9135,6 +9135,7 @@ in
   openjdk_headless = openjdk8_headless;
 
   jdk8 = openjdk8;
+  jdk8_headless = openjdk8_headless;
   jre8 = openjdk8.jre;
   jre8_headless = openjdk8_headless.jre;
 
@@ -10408,7 +10409,7 @@ in
   bazel_0_26 = callPackage ../development/tools/build-managers/bazel/bazel_0_26 {
     inherit (darwin) cctools;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation CoreServices Foundation;
-    buildJdk = jdk8;
+    buildJdk = jdk8_headless;
     buildJdkName = "jdk8";
     runJdk = jdk11_headless;
     stdenv = if stdenv.cc.isClang then llvmPackages_6.stdenv else stdenv;
@@ -10417,7 +10418,7 @@ in
   bazel_0_29 = callPackage ../development/tools/build-managers/bazel/bazel_0_29 {
     inherit (darwin) cctools;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation CoreServices Foundation;
-    buildJdk = jdk8;
+    buildJdk = jdk8_headless;
     buildJdkName = "jdk8";
     runJdk = jdk11_headless;
     stdenv = if stdenv.cc.isClang then llvmPackages_6.stdenv else stdenv;
@@ -10427,7 +10428,7 @@ in
   bazel_1 = callPackage ../development/tools/build-managers/bazel/bazel_1 {
     inherit (darwin) cctools;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation CoreServices Foundation;
-    buildJdk = jdk8;
+    buildJdk = jdk8_headless;
     buildJdkName = "jdk8";
     runJdk = jdk11_headless;
     stdenv = if stdenv.cc.isClang then llvmPackages_6.stdenv else stdenv;
@@ -10437,7 +10438,7 @@ in
   bazel_3 = callPackage ../development/tools/build-managers/bazel/bazel_3 {
     inherit (darwin) cctools;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation CoreServices Foundation;
-    buildJdk = jdk8;
+    buildJdk = jdk8_headless;
     buildJdkName = "jdk8";
     runJdk = jdk11_headless;
     stdenv = if stdenv.cc.isClang then llvmPackages_6.stdenv else stdenv;

